### PR TITLE
OCD-3902: Insert CHPL Product Number History

### PR DESCRIPTION
**Pull request by kekey1**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/CertificationResultContainsSvapActivityExplorer.java [hunk 1]
```diff
@@ -41,7 +43,7 @@ public CertificationResultContainsSvapActivityExplorer(ActivityDAO activityDao,
 
     @Override
     @Transactional
-    public ActivityDTO getActivity(ListingActivityQuery query) {
+    public List<ActivityDTO> getActivities(ListingActivityQuery query) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/CertificationResultContainsSvapActivityExplorer.java:43]: Does nothing call the original version of this method any more?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/RealWorldTestingEligibilityActivityExplorer.java [hunk 2]
```diff
@@ -30,31 +33,36 @@ public RealWorldTestingEligibilityActivityExplorer(ActivityDAO activityDao) {
...
                     activityReturn = currActivity;
                 }
             }
         }
-        return activityReturn;
+
+        if (activityReturn == null) {
+            return Collections.emptyList();
+        }
+        return Stream.of(activityReturn).toList();
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/RealWorldTestingEligibilityActivityExplorer.java]: Should we regression test whatever this RWT explorer used this call for? It looks like there might be multiple activities for a given day (time period?) and we should make sure that didn't break anything, right?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/SvapNoticeUrlLastUpdateActivityExplorer.java [hunk 3]
```diff
@@ -75,8 +78,9 @@ public ActivityDTO getActivity(ListingActivityQuery query) {
 
         if (svapNoticeLastUpdateActivity == null) {
             LOGGER.warn("Unable to determine when listing " + svapQuery.getListingId() + " set SVAP Notice URL to " + svapQuery.getSvapNoticeUrl());
+            return Collections.emptyList();
         }
-        return svapNoticeLastUpdateActivity;
+        return Stream.of(svapNoticeLastUpdateActivity).toList();
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/SvapNoticeUrlLastUpdateActivityExplorer.java]: Similar question re: SVAP whatever this is

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/CertificationResultContainsSvapActivityExplorer.java (hunk 1 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/CertificationResultContainsSvapActivityExplorer.java:43]: No... but I am going to think about if there is a better way to make this code change. This is the first activity explorer that has needed to return more than one activity.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/RealWorldTestingEligibilityActivityExplorer.java (hunk 2 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/RealWorldTestingEligibilityActivityExplorer.java]: Feel free to look at the test cases. I made notes about these changes during my development and included regression tests for all of them.

**Review by tmy1313 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**